engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala

```diff
@@ -187,24 +188,25 @@ object PackageRepository {
 
     /** The mapping containing loaded component groups.
       *
+      * It should be modified and read only from within synchronized sections.
       * The component mapping is added to the collection after ensuring that the
       * corresponding library was loaded.
       */
     private val loadedComponents
-      : collection.concurrent.TrieMap[LibraryName, ComponentGroups] = {
+      : collection.mutable.Map[LibraryName, ComponentGroups] = {
       val builtinsName = LibraryName(Builtins.NAMESPACE, Builtins.PACKAGE_NAME)
-      collection.concurrent.TrieMap(builtinsName -> ComponentGroups.empty)
+      collection.mutable.LinkedHashMap(builtinsName -> ComponentGroups.empty)
```

Comment:
Replacing _concurrent_ with _mutable_. Is concurrent access still safe? Or is concurrent access prevented? I see bunch of `this.synchronized` below, so I assume the answer is yes, concurrent access is still safe.

Reply:
Yes, the usages are synchronized, and there's a  comment there